frontend/src/components/contracts/ContractDetails.tsx

```diff
@@ -0,0 +1,127 @@
+import moment from "moment";
+
+import React from "react";
+
+import { Row, Col } from "react-bootstrap";
+
+import ContractsApi from "../../libraries/explorer-wamp/contracts";
+
+import CardCell from "../utils/CardCell";
+import TransactionLink from "../utils/TransactionLink";
+
+interface Props {
+  accountId: string;
+}
+
+interface State {
+  locked?: boolean;
+  transactionHash?: string;
+  timestamp?: number;
+}
+
+export default class extends React.Component<Props, State> {
+  state: State = {
+    locked: true
+  };
+
+  getContractInfo = async () => {
+    new ContractsApi()
+      .getContractInfo(this.props.accountId)
+      .then(contractInfo => {
+        if (contractInfo) {
+          this.setState({
+            transactionHash: contractInfo.transactionHash,
+            timestamp: contractInfo.timestamp,
+            locked: contractInfo.accessKeys.some(
+              (key: any) => key["access_key"]["permissions"] === "FullAccess"
```

Comment:
1. This is an inverted logic here. The "locked" contact is one that nobody can ever change, so there is no other permission but FunctionCall.
2. Read my comments about the difference "no FullAccess" vs "no permissions except FunctionCalls" above.